k8s_handle/__init__.py

```diff
@@ -15,19 +15,125 @@
 from k8s_handle.k8s.deprecation_checker import ApiDeprecationChecker, DeprecationError
 from k8s_handle.k8s.resource import Provisioner, ProvisioningError
 
+COMMAND_DEPLOY = 'deploy'
+COMMAND_DESTROY = 'destroy'
+
 log = logging.getLogger(__name__)
 logging.basicConfig(level=settings.LOG_LEVEL, format=settings.LOG_FORMAT, datefmt=settings.LOG_DATE_FORMAT)
 
+
+def handler_deploy(args):
+    _handler_deploy_destroy(args, COMMAND_DEPLOY)
+
+
+def handler_destroy(args):
+    _handler_deploy_destroy(args, COMMAND_DESTROY)
+
+
+def handler_apply(args):
+    _handler_apply_delete(args, COMMAND_DEPLOY)
+
+
+def handler_delete(args):
+    _handler_apply_delete(args, COMMAND_DESTROY)
+
+
+def handler_template(args):
+    context = config.load_context_section(args.get('section'))
+    templating.Renderer(
+        settings.TEMPLATES_DIR,
+        args.get('tags'),
+        args.get('skip_tags')
+    ).generate_by_context(context)
+
+
+def _handler_deploy_destroy(args, command):
+    context = config.load_context_section(args.get('section'))
+    resources = templating.Renderer(
+        settings.TEMPLATES_DIR,
+        args.get('tags'),
+        args.get('skip_tags')
+    ).generate_by_context(context)
+
+    if args.get('dry_run'):
+        return
+
+    _handler_common(
+        command,
+        resources,
+        config.PriorityEvaluator(args, context, os.environ),
+        args.get('use_kubeconfig'),
+        args.get('sync_mode'),
+        args.get('show_logs')
+    )
+
+
+def _handler_apply_delete(args, command):
+    _handler_common(
+        command,
+        [os.path.join(settings.TEMP_DIR, args.get('resource'))],
+        config.PriorityEvaluator(args, {}, os.environ),
+        args.get('use_kubeconfig'),
+        args.get('sync_mode'),
+        args.get('show_logs')
+    )
+
+
+def _handler_common(command, resources, priority_evaluator, use_kubeconfig, sync_mode, show_logs):
```

Comment:
Thought about moving handlers into other file, but decided not to do this, "good enough".
Assigning subcommands handlers, DRY generalization.